### PR TITLE
SDBELGA-152 Implement OpenID Connect authentication

### PR DESCRIPTION
**Pull request by idsvn**
Belga uses KeyCloak server for Single Sign On.

### CODE REVIEW

File: server/requirements.txt [hunk 1]
```diff
@@ -7,3 +7,5 @@ oauth2client==4.1.3
 -e git+git://github.com/superdesk/superdesk-core.git@develop#egg=Superdesk-Core
 -e git+git://github.com/superdesk/superdesk-analytics.git@master#egg=superdesk-analytics
 -e git+git://github.com/superdesk/superdesk-planning.git@master#egg=superdesk-planning
+
+flask-oidc==1.4.0
```
Comment (idsvn) [inline server/requirements.txt]: When we put `flask-oidc` at the begin of file, I conflict with some library in Superdesk-Core and cannot install, So we add it in the end of file.

File: server/belga/auth/__init__.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+class AuthResource(Resource):
+    schema = {
+        'token': {
+            'type': 'string'
+        },
+        'user': Resource.rel('users', True)
+    }
+    resource_methods = ['POST']
+    item_methods = ['GET', 'DELETE']
+    public_methods = ['POST', 'DELETE']
```
Comment (petrjasek) [inline server/belga/auth/__init__.py]: not sure if there should be public delete

File: server/settings.py [hunk 3]
```diff
@@ -221,3 +223,11 @@
 KEYWORDS_ADD_MISSING_ON_PUBLISH = True
 MACROS_MODULE = env('MACROS_MODULE', 'belga.macros')
 
+SECRET_KEY = 'SomethingNotEntirelySecret'
+OIDC_CLIENT_SECRETS = env('OIDC_CLIENT_SECRETS', 'client_secrets.json')
```
Comment (petrjasek) [inline server/settings.py]: would be good to use env variables for all secrets, also for client secret parts:
```py
SECRET_KEY = env('SECRET_KEY')
OIDC_CLIENT_SECRETS = {
    'web': {
        'client_id: env('OIDC_CLIENT_ID'),
        'client_secret': env('OIDC_CLIENT_SECRET'),
    },
}
```
or maybe just define those `OIDC_CLIENT_SECRET` etc and construct the config using these in init_app:
```py
def init_app(app):
    app.config.setdefault('OIDC_CLIENT_SECRETS', {
        'web': {
             'client_secret: app.config.get('OIDC_CLIENT_SECRET'),
             ...,
        },
    })
```

File: server/client_secrets.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (petrjasek) [inline server/client_secrets.json]: this file shouldn't be in git, you can add it to `.gitignore`

File: server/settings.py (hunk 3 above)
Comment (idsvn) [reply server/settings.py]: Unfortunately, there is no release from commit https://github.com/puiterwijk/flask-oidc/commit/de9584677fbed8f8688eab60145744cb353915ea which support loading `CLIENT_SECRET` from dict, we are considering [Flask-OIDC-ex](https://github.com/larsw/flask-oidc-ex), if it's working fine, we will update accordingly.

**Comment by idsvn** [comment]
Keycloak Auth will be implemented in `superdesk-core` so close this one.

**Pull request closed without merging**